usmqe/api/tendrlapi/authentication.py

```diff
@@ -0,0 +1,60 @@
+"""
+Tendrl REST API.
+"""
+
+import requests
+
+import pytest
+
+import json
+
+from usmqe.api.tendrlapi.common import TendrlApi
+
+LOGGER = pytest.get_logger("tendrlapi.authentication", module=True)
+
+
+class Authentication(TendrlApi):
+    """ Main class for interact with REST API - authentication."""
+
+    def __init__(self):
+        self.username = None
+        self.access_token = None
+
+    def login(self, username, password, asserts_in=None):
+        """ Login user.
+
+        Name:        "login",
+        Method:      "GET",
+        Pattern:     "login",
+
+        Args:
+            username: name of user that is going logged in
+            password: password for username
+            asserts_in: assert values for this call and this method
+        """
+        pattern = "login"
+        post_data = {"username": username, "password": password}
+        request = requests.post(
+            pytest.config.getini("usm_api_url") + pattern, data=json.dumps(post_data))
```

Comment:
Another note for all our methods. They are all the same or very similar. Some request.post/request.put/request.get or request.delete, call print_req_info, check_response and return response of the request. As I see it we could have just two general methods, one with data other without, and use them. With that the pattern logic make sense.